File: 1004-max-consecutive-ones-iii/1004-max-consecutive-ones-iii.cpp
class Solution {
public:
    int longestOnes(vector<int>& nums, int k) {
    int left = 0; 
    int zeroCount = 0;  
    int window = 0;  
    
    for (int right = 0; right < nums.size(); right++) {
        if (nums[right] == 0) zeroCount++;
        while (zeroCount > k) 
        {
            if (nums[left] == 0)
                zeroCount--;
            left++;
        }
        
        window = max(window, right - left + 1);
    }
    
    return window;
    }
     
};

File: 1004-max-consecutive-ones-iii/1004-max-consecutive-ones-iii.js
/**
 * @param {number[]} nums
 * @param {number} k
 * @return {number}
 */
var longestOnes = function(nums, k) {
    let window = 0, lastZero = 0,counteZero = 0;
    for(let i = 0; i < nums.length ; i++)
     {
        if (!nums[i]) counteZero++;
        while(counteZero > k)
        {
            if (!nums[lastZero]) counteZero--;
            lastZero++;
        }
        window = Math.max(window,i - lastZero + 1);
     }
    return window;
};